tools/rebuild-transfer-backlog

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+
+# This file is part of the Archivematica development tools.
+#
+# Copyright 2010-2016 Artefactual Systems Inc. <http://artefactual.com>
+#
+# Archivematica is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Archivematica is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Archivematica.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import print_function
+
+import ConfigParser
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+import time
+import xml.etree.ElementTree as ElementTree
+
+import django
+
+import elasticSearchFunctions
+from main.models import File, Transfer
+import storageService
+
+sys.path.append('/usr/lib/archivematica/MCPClient/clientScripts')
+import sanitizeNames
+
+
+django.setup()
+
+
+def get_file_checksum(path):
+    p = subprocess.Popen(['sha256sum', path], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    out, err = p.communicate()
+    return out.split(' ')[0]
+
+
+def process_transfer(es_client, path):
+    transfer_dir = os.path.basename(path)
+    uuid = transfer_dir[-36:]
+
+    # Parse METS data
+    tree = ElementTree.parse(os.path.join(path, 'metadata/submissionDocumentation/METS.xml'))
+    root = tree.getroot()
+
+    # Attempt to insert row in MySQL Transfers table if it doesn't already exist
+    try:
+        Transfer.objects.create(uuid=uuid, currentlocation=os.path.join('%sharedPath%www/AIPsStore/transferBacklog/originals/', transfer_dir))
```

Comment:
I'd recommend against it. I'm not sure what would break, but I suspect something would. :)

You could ask for the shared directory to be passed in as a parameter, or read it from the serverConfig.conf file.